cmd/root.go

```diff
@@ -50,10 +55,52 @@ func NewMethodAws(version string) *MethodAws {
 	return &methodAws
 }
 
+// Helper function to set up common configurations
+func (a *MethodAws) setupCommonConfig(cmd *cobra.Command, outputFormat string, outputFile string, authed bool) error {
+	var err error
+
+	if authed {
+		awsConfig, err := awsconfig.LoadDefaultConfig(cmd.Context())
+		if err != nil {
+			return err
+		}
+		a.AwsConfig = &awsConfig
+		a.RootFlags.Regions, err = common.GetAWSRegions(cmd.Context(), *a.AwsConfig, a.RootFlags.Regions)
+		if err != nil || len(a.RootFlags.Regions) == 0 {
+			a.OutputSignal.Status = 401
+			a.OutputSignal.ErrorMessage = aws.String("No valid AWS regions found or specified")
+			return nil
+		} else {
+			a.AwsConfig.Region = a.RootFlags.Regions[0]
+		}
+	} else {
```

Comment:
This feels wrong. You're setting up a totally separate logger if and only if it's unauthenticated. When you rip the log out of the function and instead pass a context, you'll want that context to be set up with the logger already (ie. move line 95 to the top of this method)